### PR TITLE
SISRP-4846 - Implemented Browsersync

### PR DESCRIPTION
**Pull request by winstonkim**
https://jira.berkeley.edu/browse/SISRP-4846
https://jira.ets.berkeley.edu/jira/browse/CLC-5522

### CODE REVIEW

**Comment by christianv** [comment]
@winstonkim can you have a look at the travis failure?

File: gulpfile.js [hunk 1]
```diff
@@ -29,11 +29,16 @@
...
+  // Initialize BrowserSync
+  var browserSync = require('browser-sync').create();
+
   // Base options for the command line
   var baseOptions = {
-    string: 'env',
-    default: {
-      env: process.env.RAILS_ENV || 'development'
+    'string': 'env',
+    'boolean': 'bs',
```
Comment (christianv) [inline gulpfile.js]: @winstonkim can we rename this to `browsersync` instead?

File: gulpfile.js [hunk 2]
```diff
@@ -115,6 +124,18 @@
   };
 
   /**
+   * BrowserSync task
+   *   Initialized BrowserSync
+   */
+  gulp.task('browser-sync', ['css', 'browserify'], function() {
+    if (useBrowserSync){
```
Comment (christianv) [inline gulpfile.js]: @winstonkim `jscs` will probably complain about not having a space after `)`

**Comment by christianv** [comment]
lgtm - @paulkerschen / @johncrossman or @raydavis mind having a look at this as well?

**Comment by paulkerschen** [comment]
@christianv It looks like Browsersync will be on-by-default in the test and testext environments as well as development. Do we want this? Or might it be better to whitelist the development environment only?

**Comment by christianv** [comment]
@paulkerschen thanks, always good to have an extra pair of eyes on it.

@winstonkim can you follow up on @paulkerschen's comment? Probably better to make this `development` only.

**Comment by winstonkim** [comment]
@paulkerschen @christianv ok, implemented it so that browsersync can only turn on during development mode only.

**Comment by christianv** [comment]
@winstonkim @paulkerschen lgtm - let's see how developers like this change.

**Pull request merged**